src/chat/GalleryScreen.tsx

```diff
@@ -0,0 +1,175 @@
+import React, { useEffect, useRef, useState } from 'react';
+import {
+  View,
+  Text,
+  FlatList,
+  StyleSheet,
+  TouchableOpacity,
+  Dimensions,
+} from 'react-native';
+import FastImage from 'react-native-fast-image';
+import { FirestoreServices } from '../services/firebase';
+import { MessageTypes, type MediaFile } from '../interfaces';
+import ThumbnailVideoPlayer from './components/camera/ThumbnailVideoPlayer';
+import SelectedViewModal from './components/camera/SelectedViewModal';
+type MediaItem = {
+  item: MediaFile;
+  index: number;
+};
+
+const { width } = Dimensions.get('window');
+
+interface GalleryModalProps {
+  renderCustomHeader?: () => JSX.Element;
+  renderCustomMedia?: ({ item, index }: MediaItem) => JSX.Element | null;
+  renderCustomFile?: () => JSX.Element;
+  renderCustomLink?: () => JSX.Element;
+}
+
+export const GalleryScreen: React.FC<GalleryModalProps> = ({
+  renderCustomHeader,
+  renderCustomMedia,
+  renderCustomFile,
+  renderCustomLink,
+}) => {
+  const [activeTab, setActiveTab] = useState('Media');
+  const [media, setMedia] = useState<MediaFile[]>([]);
+  const firebaseInstance = useRef(FirestoreServices.getInstance()).current;
+  const [mediaSelected, setMediaSelected] = useState<MediaFile>();
+
+  useEffect(() => {
+    if (activeTab === 'Media') {
+      const loadImages = async () => {
+        const urls = await firebaseInstance.getMediaFilesByConversationId();
+        setMedia(urls);
+      };
+      loadImages();
+    }
+  }, [activeTab, firebaseInstance]);
+
+  const renderImage = ({ item, index }: MediaItem) => {
+    if (renderCustomMedia) return renderCustomMedia({ item, index });
+    return (
+      <TouchableOpacity onPress={() => setMediaSelected(item)}>
+        {item.type === MessageTypes.video ? (
+          <ThumbnailVideoPlayer videoUrl={item.path} />
+        ) : (
+          <FastImage source={{ uri: item.path }} style={styles.image} />
+        )}
+      </TouchableOpacity>
+    );
+  };
+
+  const renderHeader = (): JSX.Element => {
+    if (renderCustomHeader) return renderCustomHeader();
+    return (
+      <View style={styles.tabContainer}>
+        {['Media', 'File', 'Link'].map((tab) => (
+          <TouchableOpacity
+            key={tab}
+            style={[styles.tab, activeTab === tab && styles.activeTab]}
+            onPress={() => setActiveTab(tab)}
+          >
+            <Text
+              style={[
+                styles.tabText,
+                activeTab === tab && styles.activeTabText,
+              ]}
+            >
+              {tab.toUpperCase()}
+            </Text>
+            {activeTab === tab && <View style={styles.tabIndicator} />}
+          </TouchableOpacity>
+        ))}
+      </View>
+    );
+  };
+
+  const renderContent = () => {
+    switch (activeTab) {
+      case 'Media':
```

Comment:
Do not use magic string and make it editable.

Reply:
ok I updated